plugins/nexusoperations/workflow/commands_test.go

```diff
@@ -0,0 +1,354 @@
+	"go.temporal.io/server/common/dynamicconfig"
+	"go.temporal.io/server/plugins/nexusoperations"
+	opsworkflow "go.temporal.io/server/plugins/nexusoperations/workflow"
+	"go.temporal.io/server/service/history/hsm"
+	"go.temporal.io/server/service/history/tests"
+	"go.temporal.io/server/service/history/workflow"
+	"google.golang.org/protobuf/types/known/durationpb"
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+type commandValidator struct {
+	maxPayloadSize int
+}
+
+func (v commandValidator) IsValidPayloadSize(size int) bool {
+	return size <= v.maxPayloadSize
+}
+
+type testContext struct {
+	ms              workflow.MutableState
+	scheduleHandler workflow.CommandHandler
+	cancelHandler   workflow.CommandHandler
+	history         *historypb.History
+}
+
+var defaultConfig = &nexusoperations.Config{
+	Enabled:                 dynamicconfig.GetBoolPropertyFnFilteredByNamespace(true),
+	MaxOperationNameLength:  dynamicconfig.GetIntPropertyFilteredByNamespace(5),
+	MaxConcurrentOperations: dynamicconfig.GetIntPropertyFilteredByNamespace(2),
+}
+
+func newTestContext(t *testing.T, cfg *nexusoperations.Config) testContext {
+	chReg := workflow.NewCommandHandlerRegistry()
+	require.NoError(t, opsworkflow.RegisterCommandHandlers(chReg, cfg))
+	smReg := hsm.NewRegistry()
+	require.NoError(t, workflow.RegisterStateMachine(smReg))
+	require.NoError(t, nexusoperations.RegisterStateMachines(smReg))
+	node, err := hsm.NewRoot(smReg, workflow.StateMachineType.ID, nil, make(map[int32]*persistencespb.StateMachineMap))
+	require.NoError(t, err)
+	ms := workflow.NewMockMutableState(gomock.NewController(t))
+	ms.EXPECT().HSM().Return(node).AnyTimes()
+	lastEventID := int64(4)
+	history := &historypb.History{}
+	ms.EXPECT().AddHistoryEvent(gomock.Any(), gomock.Any()).DoAndReturn(func(t enumspb.EventType, setAttrs func(he *historypb.HistoryEvent)) *historypb.HistoryEvent {
+		e := &historypb.HistoryEvent{
+			Version:   1,
+			EventId:   lastEventID,
+			EventTime: timestamppb.Now(),
+		}
+		lastEventID++
+		setAttrs(e)
+		history.Events = append(history.Events, e)
+		return e
+	}).AnyTimes()
+	ms.EXPECT().GetNamespaceEntry().Return(tests.GlobalNamespaceEntry).AnyTimes()
+	ms.EXPECT().GetExecutionInfo().Return(&persistencespb.WorkflowExecutionInfo{}).AnyTimes()
+	scheduleHandler, ok := chReg.Handler(enumspb.COMMAND_TYPE_SCHEDULE_NEXUS_OPERATION)
+	require.True(t, ok)
+	cancelHandler, ok := chReg.Handler(enumspb.COMMAND_TYPE_REQUEST_CANCEL_NEXUS_OPERATION)
+	require.True(t, ok)
+
+	return testContext{
+		ms:              ms,
+		history:         history,
+		scheduleHandler: scheduleHandler,
+		cancelHandler:   cancelHandler,
+	}
+}
+
+func TestHandleScheduleCommand(t *testing.T) {
+	t.Run("feature disabled", func(t *testing.T) {
+		tcx := newTestContext(t, &nexusoperations.Config{
+			Enabled: dynamicconfig.GetBoolPropertyFnFilteredByNamespace(false),
+		})
+		err := tcx.scheduleHandler(tcx.ms, commandValidator{1}, 1, &commandpb.Command{})
```

Comment:
nit: I would make this `commandValidator{maxPayloadSize: 1}` to make it a bit more readable here and elsewhere.